src/main/java/seedu/address/model/task/Task.java

```diff
@@ -447,6 +447,19 @@ public boolean isPastCurrentDateAndRecurringTask() {
         return DateTime.isOverdue(this.dateTime) && recurrence.isRecurring();
     }
 
+    /**
+     * Returns true if both tasks have the same description.
+     * This defines a weaker notion of equality between two tasks.
+     */
+    public boolean isSameTask(Task otherTask) {
```

Comment:
We allow multiple tasks with same descriptions. But date, time and description cannot be all the same. Different recurrences still considered "same task" since no point in a daily/weekly/monthly task of the same date time to exist at once.

Reply:
I was thinking maybe we could have another error message for duplicate tasks with different recurrences- maybe append the current MESSAGE_DUPLICATE_TASK with a suggestion to edit the task directly? I feel this would benefit user experience.

Reply:
Used isSameTask() to check if `Description`, `Date` and `Time` are the same for "same" tasks.
Appended suggestion as stated above.